src/accumulator/proof.rs

```diff
@@ -305,21 +330,27 @@ impl Proof {
 
                 // If the next node is not my sibling, the hash must be passed inside the proof
                 if let Some(next_proof_hash) = hashes_iter.next() {
-                    let hash = if util::is_left_niece(pos) {
-                        types::parent_hash(&hash, next_proof_hash)
-                    } else {
-                        types::parent_hash(next_proof_hash, &hash)
-                    };
+                    if hash != sha256::Hash::default() {
+                        let hash = if util::is_left_niece(pos) {
+                            types::parent_hash(&hash, next_proof_hash)
+                        } else {
+                            types::parent_hash(next_proof_hash, &hash)
+                        };
 
-                    let next_to_prove = util::parent(pos, total_rows);
+                        let next_to_prove = util::parent(pos, total_rows);
 
-                    Proof::sorted_push(&mut nodes, (next_to_prove, hash));
+                        Proof::sorted_push(&mut nodes, (next_to_prove, hash));
+                        continue;
+                    } else {
+                        // If none of the above, push a nil hash upwards
```

Comment:
Technically not a nil right? Maybe just "push a default hash upwards"

Reply:
Yeah, it's the default hash. You can't have a nil in safe rust outside an option.